pocs/utils/google/storage.py

```diff
@@ -88,3 +91,239 @@ def upload(self, local_path, remote_path=None):
+                    except TypeError as e:
+                        warn('parse_line must be callable')
+                    continue
+
+                # Get key=value pairs (skip COMMENTS and HISTORY)
+                if item_string.find('=') > 0:
+                    k, v = item_string.split('=')
+
+                    # Remove FITS comment
+                    if ' / ' in v:
+                        v = v.split(' / ')[0]
+
+                    v = v.strip()
+
+                    # Cleanup and discover type in dumb fashion
+                    if v.startswith("'") and v.endswith("'"):
+                        v = v.replace("'", "").strip()
+                    elif v.find('.') > 0:
+                        v = float(v)
+                    elif v == 'T':
+                        v = True
+                    elif v == 'F':
+                        v = False
+                    else:
+                        v = int(v)
+
+                    headers[k.strip()] = v
+
+            i += 1
+
+        return headers
+
+
+def upload_observation_to_bucket(pan_id,
+                                 dir_name,
+                                 include_files='*.fz',
+                                 bucket='panoptes-survey',
+                                 **kwargs):
+    """Upload an observation directory to google cloud storage.
+
+    This is a convenience function for bulk uploading an observation folder to a
+    bucket. This assumes that observations are placed within `/images/fields`
+    and follow the normal naming convention for observations.
+
+    Note:
+        This requires that the command line utility `gsutil` be installed
+        and that authentication has properly been set up.
+
+    Args:
+        pan_id (str): A string representing the unit id, e.g. PAN001.
+        dir_name (str): Full path to directory.
+        include_files (str, optional): Filename filter, defaults to
+            compressed FITS files '.fz'.
+        bucket (str, optional): The bucket to place the images in, defaults
+            to 'panoptes-survey'.
+        **kwargs: Optional keywords: verbose
+    """
+    assert os.path.exists(dir_name)
+    assert re.match(r'PAN\d\d\d', pan_id) is not None
+
+    verbose = kwargs.get('verbose', False)
+
+    def _print(msg):
+        if verbose:
+            print(msg)
+
+    _print("Uploading {}".format(dir_name))
+
+    gsutil = shutil.which('gsutil')
+    assert gsutil is not None
+
+    img_path = os.path.join(dir_name, include_files)
+    if glob(img_path):
+        # Get just the observation path
+        field_dir = dir_name.split('/fields/')[-1]
+        remote_path = os.path.normpath(os.path.join(
+            bucket,
+            pan_id,
+            field_dir
+        ))
+
+        # normpath strips the trailing slash so add here so files go in directory
+        destination = 'gs://{}/'.format(remote_path)
+        run_cmd = [gsutil, '-mq', 'cp', '-r', img_path, destination]
```

Comment:
I just tried running gsutil locally and it failed with:
```
ERROR: Python 3 and later is not compatible with the Google Cloud SDK. Please use Python version 2.7.x.

If you have a compatible Python interpreter installed, you can use it by setting
the CLOUDSDK_PYTHON environment variable to point to it.
```
How is it working for you?

Reply:
Oh, hm. Perhaps I had this set up earlier under python 2.7. Let me look into my setup. In my head the gsutil had moved up to python 3.

Reply:
So gcloud and gsutil come from the same place and both require python 2.7. The gsutil is nice for uploading because it transparently handles parallel uploads while the python3 client library does not. We can use the service key to authenticate with either. If we wanted to completely avoid python2.7 we could just use the client libraries, just missing that parallel ability, which is very nice to have.

The gcloud and gsutil can come from the ubuntu package manager, so part of #578 .